github/dependency_graph_snapshots.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// DependencyGraphSnapshotResolvedDependencyRelationship represents whether the dependency is requested directly by the manifest or is a dependency of another dependency.
+//
+// Can have the following values:
+//   - "direct": indicates that the dependency is requested directly by the manifest.
+//   - "indirect": indicates that the dependency is a dependency of another dependency.
+type DependencyGraphSnapshotResolvedDependencyRelationship string
+
+// DependencyGraphSnapshotResolvedDependencyScope represents whether the dependency is required for the primary build artifact or is only used for development.
+//
+// Can have the following values:
+//   - "runtime": indicates that the dependency is required for the primary build artifact.
+//   - "development": indicates that the dependency is only used for development.
+type DependencyGraphSnapshotResolvedDependencyScope string
+
+// DependencyGraphSnapshotCreationResult represents the snapshot creation result.
+//
+// Can have the following values:
+//   - "SUCCESS": indicates that the snapshot was successfully created and the repository's dependencies were updated.
+//   - "ACCEPTED": indicates that the snapshot was successfully created, but the repository's dependencies were not updated.
+//   - "INVALID": indicates that the snapshot was malformed.
+type DependencyGraphSnapshotCreationResult string
+
+type DependencyGraphSnapshotResolvedDependency struct {
```

Comment:
Each new struct type needs a Godoc-style comment that starts with its name and make a complete statement about its purpose, ending with a period, so that the auto-generated documentation reads nicely.